bm-persona/Map/MapViewController.swift

```diff
@@ -167,19 +198,44 @@ extension MapViewController: MKMapViewDelegate {
             annotationView.annotation = marker
             annotationView.image = marker.type.icon()
             return annotationView
+        } else if let searchAnnotation = annotation as? SearchAnnotation,
+            // create new pin on map for searched item
+            let annotationView = mapView.dequeueReusableAnnotationView(withIdentifier: MKMapViewDefaultAnnotationViewReuseIdentifier) as? MKMarkerAnnotationView {
+            annotationView.annotation = searchAnnotation
+            annotationView.glyphImage = searchAnnotation.icon()
+            annotationView.contentMode = .scaleToFill
+            annotationView.markerTintColor = searchAnnotation.color()
+            annotationView.glyphTintColor = .white
+            return annotationView
         }
         return MKAnnotationView()
     }
     
     func mapView(_ mapView: MKMapView, didSelect view: MKAnnotationView) {
-        markerDetail.marker = view.annotation as? MapMarker
-        drawerContainer?.moveDrawer(to: .hidden, duration: 0.2)
+        if let annotation = view.annotation as? MapMarker {
+            markerDetail.marker = annotation
+            // hide either search detail or drawer
+            if detailViewController != nil {
+                detailViewController!.view.isHidden = true
+            } else {
+                drawerContainer?.moveDrawer(to: .hidden, duration: 0.2)
+            }
+        }
     }
     
     func mapView(_ mapView: MKMapView, didDeselect view: MKAnnotationView) {
-        markerDetail.marker = nil
-        drawerContainer?.moveDrawer(to: .collapsed, duration: 0.2)
+        if (view.annotation as? MapMarker) != nil {
+            markerDetail.marker = nil
+            // show either search detail or drawer
+            if detailViewController != nil {
+                detailViewController!.view.isHidden = false
+                mapView.selectAnnotation(searchAnnotation!, animated: true)
```

Comment:
strange problem with the obvious fix you suggested: map marker is deselected, which selects the searchAnnotation, but then immediately the second map marker is selected. however, only one annotation can be selected at a time. what happens is after searchAnnotation is selected, the second map marker being selected somehow removes searchAnnotation from the array of selected annotations without deselecting it. the searchAnnotation grows big, but calling deselect on it doesn't do anything because it's no longer in the array of selected annotations. my fix was asynchronously checking if markerDetail.marker is nil before showing the detail view. not sure if there's a better way to do it.